iceoryx_posh/test/moduletests/test_posh_runtime.cpp

```diff
@@ -493,3 +493,46 @@ TEST_F(PoshRuntime_test, CreateRunnableReturnValue)
     /// @todo I am passing runnableDeviceIdentifier as 1, but it returns 0, is this expected?
     // EXPECT_EQ(runnableDeviceIdentifier, runableData->m_runnableDeviceIdentifier);
 }
+
+class PoshRuntimeTestAccess : public PoshRuntime
+{
+  public:
+    using PoshRuntime::factory_t;
+    using PoshRuntime::setRuntimeFactory;
+
+    PoshRuntimeTestAccess(const iox::ProcessName_t& s)
+        : PoshRuntime(s)
+    {
+    }
+
+    static PoshRuntimeTestAccess*& getTestRuntime()
+    {
+        static PoshRuntimeTestAccess* testRuntime = nullptr;
+        return testRuntime;
+    }
+};
+
+namespace
+{
+bool callbackWasCalled = false;
+PoshRuntime& testFactory(const iox::ProcessName_t&)
+{
+    callbackWasCalled = true;
+    return *PoshRuntimeTestAccess::getTestRuntime();
+}
+} // namespace
+
+TEST_F(PoshRuntime_test, SetValidRuntimeFactorySucceeds)
+{
+    PoshRuntimeTestAccess::setRuntimeFactory(testFactory);
+    PoshRuntimeTestAccess::getInstance("/instance");
+
+    EXPECT_TRUE(callbackWasCalled);
```

Comment:
Maybe narrow down that the callback was called in the getInstance call (and not by setting the factory) by setting ``callbackWasCalled=false;`` right before the getInstance call.

Reply:
You are right but I would prefer setting this in the `SetUp` call of the test fixture like:
```cpp
class PoshRuntimeTestAccess : public PoshRuntime
{
  public:
     void SetUp() {
         callbackWasCalled = false;
     }
};
```